src/pages/LandingPage.jsx

```diff
@@ -0,0 +1,84 @@
+// ============================================================================================
+// IMPORTS
+// ============================================================================================
+
+import { useContext, useState } from "react";
+import ResponsiveGrid from "../components/ResponsiveGrid";
+import { CoursesContext } from "../contexts/CoursesContext";
+import CourseCard from "../components/CourseCard";
+import Course from "../components/Course";
+import { useNavigate } from "react-router-dom";
+
+const serverURL = import.meta.env.VITE_SERVER_URL_ROOT;
+
+console.assert(
+  serverURL?.length > 0,
+  'Server URL not specified -- add "VITE_SERVER_URL_ROOT=<url>" to .env'
+);
+
+function LandingPage() {
+  // CONTEXT AND STATE
+  // context will have all the courses that will be displayed in the landing page
+  const { coursesInfo, setCoursesInfo } = useContext(CoursesContext);
+  // Course has the information for the course that would like to be displayed
+  const [course, setCourse] = useState(null);
+  // controlled input state that will help to filter the courses that are displayed
+  const [searchText, setSearchText] = useState("");
```

Comment:
Ensure that searchText is reset or cleared when appropriate, such as after performing a search or resetting the courses.

Reply:
after performing a search the text is left there as an indication to the user of what are they looking at.
I made some changes to ensure that after reseting the search, the input is cleared.